.github/workflows/css_check.py

```diff
@@ -0,0 +1,147 @@
+"""Check TypeScript files for CSS violations and embedded CSS."""
+
+import argparse
+import os
+import re
+import sys
+
+
+def check_embedded_css(content: str) -> list:
+    """
+    Check for embedded CSS in the content.
+
+    Args:
+        content: The content of the file to check.
+
+    Returns:
+        A list of embedded CSS violations found.
+    """
+    embedded_css_pattern = r"#([0-9a-fA-F]{3}){1,2}"  # Matches CSS color codes
+    return re.findall(embedded_css_pattern, content)
+
+
+def check_files(
+    directory: str, exclude_files: list, exclude_directories: list, allowed_css_patterns: list
+) -> tuple:
+    """
+    Check TypeScript files for CSS violations and print correct CSS imports.
+
+    Args:
+        directory: The directory to check.
+        exclude_files: List of files to exclude from analysis.
+        exclude_directories: List of directories to exclude from analysis.
+        allowed_css_patterns: List of allowed CSS file patterns.
+
+    Returns:
+        A tuple containing lists of violations, correct CSS imports, and embedded CSS violations.
+    """
+    violations = []
+    correct_css_imports = []
+    embedded_css_violations = []
+
+    # Normalize exclude paths
+    exclude_files = set(os.path.abspath(file) for file in exclude_files)
+    exclude_directories = set(os.path.abspath(dir) for dir in exclude_directories)
+
+    for root, _, files in os.walk(directory):
+        # Skip excluded directories
+        if any(root.startswith(exclude_dir) for exclude_dir in exclude_directories):
+            continue
+
+        for file in files:
+            file_path = os.path.abspath(os.path.join(root, file))
+
+            # Skip excluded files
+            if file_path in exclude_files:
+                continue
+
+            # Process TypeScript files
+            if file.endswith((".ts", ".tsx")) and "test" not in root:
+                try:
+                    with open(file_path, "r", encoding="utf-8") as f:
+                        content = f.read()
+                except (IOError, UnicodeDecodeError) as e:
+                    print(f"Error reading file {file_path}: {e}")
+                    continue
+
+                # Check for CSS imports with an improved regex pattern
+                css_imports = re.findall(
```

Comment:
How do you handle the use of single or double quotes? What does prettier apply, if relevant?